File: src/transform.cpp
#include <cmath>
#include <ros/ros.h>
#include <tf/transform_broadcaster.h>

int main(int argc, char* argv[])
{
  ros::init(argc, argv, "transform");
  ros::NodeHandle nodeHandle;
  (void)nodeHandle;
  ros::Rate nodeRate(10);

  tf::TransformBroadcaster tfb;

  //change these depending on where the kinect is
  double offsetx = 0.25;
  double offsety = 0.0;
  double offsetz = 0.18;

  while (ros::ok())
  {
    tf::StampedTransform transform;

    transform.setOrigin( tf::Vector3(offsetx, offsety, offsetz));

    tf::Quaternion q;
    q.setEuler(0.0, 0.0, 0.0);
    transform.setRotation(q);

    transform.stamp_ = ros::Time::now(); //timestamp of transform

    transform.frame_id_ = "base_link";
    transform.child_frame_id_ = "openni_depth_frame";

    tfb.sendTransform(transform);

    ros::spinOnce();
    nodeRate.sleep();
  }

  return 0;
}

File: scripts/transform.py
#!/usr/bin/env python

import roslib; roslib.load_manifest('movement')
import rospy
import tf

def transform():
  rospy.init_node('movementtransform')
  
  broadcaster = tf.TransformBroadcaster()

  frequency = 10

  rate = rospy.Rate(frequency)
  #offset variables (change these to adjust for kinect position)
  offsetx = 0.25
  offsety = 0.0
  offsetz = 0.18

  while not rospy.is_shutdown():

    #transform = tf.StampedTransform()
    #transform.setOrigin(tf.Vector3(offsetx, offsety, offsetz))
    
    #q = tf.Quaternion
    #q.setEuler(0.0, 0.0, 0.0) #no rotation needs to be made
    #transform.setRotation(q)

    #transform.frame_id_ = "base_link"
    #transform.child_frame_id_ = "openni_depth_frame"

    frame_id = "base_link"
    child_frame_id = "openni_depth_frame"

    broadcaster.sendTransform((offsetx, offsety, offsetz), tf.transformations.quaternion_from_euler(0,0,0), rospy.Time.now(), frame_id, child_frame_id)

    rate.sleep()

if __name__ == '__main__':
  transform()
